src/server/locals.js

```diff
@@ -51,12 +57,51 @@ Locals.prototype.initSession = function() {
     }
   });
 
+  var self = this;
   Locals.session.post('Debugger.enable', (_err, _result) => {
-    Locals.session.post('Debugger.setPauseOnExceptions', { state: 'all'}, (_err, _result) => {
-    });
+    self.initialized = true;
+    updatePauseState(self.options, self.logger);
   });
 }
 
+Locals.prototype.disconnectSession = function() {
+  if (Locals.session) {
+    updatePauseState({ enabled: false }, this.logger);
+    Locals.session.disconnect();
+    Locals.session = null;
+  }
+}
+
+Locals.prototype.updateOptions = function(options) {
+  var setPauseChanged = this.options.enabled != options.enabled || this.options.uncaughtOnly != options.uncaughtOnly;
+
+  this.options = _.merge(this.options, options);
+
+  if (this.initialized && setPauseChanged) {
+    updatePauseState(this.options, this.logger);
+  }
+}
+
+function updatePauseState(options, logger) {
+  var state = pauseStateFromOptions(options);
+  console.log('setPauseOnExceptions', state);
+  Locals.session.post('Debugger.setPauseOnExceptions', { state: state}, (err, _result) => {
+    logger.error('error in setPauseOnExceptions', err);
+  });
+}
```

Comment:
I can be totally wrong here but...the purpose of the `setPauseChanged` flag is only to filter out log informations in case there was no actual change?

IMHO it can be quite frustrating while debugging sometimes you get information and sometimes you not.

I would rather send information in any case while user is calling `updateOptions()`.
So in this case I would avoid `setPauseChanged` and keep the condition like this:
```javascript
if (this.initialized) {
  updatePauseState(this.options, this.logger);
}
```

We could also return `state` as `unmodified` for this case (IMHO not necessary).

Reply:
It's not for logging. It's so `Locals.session.post('Debugger.setPauseOnExceptions', { state: state}...` which sets a new state, is only called if the state has actually changed.

Reply:
I'v updated the var to `pauseStateChanged for clarity. and added a missing `if (err) {}` around the `log.error`.

Reply:
It looks OK now! Thanks